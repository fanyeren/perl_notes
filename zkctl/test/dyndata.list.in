#!/bin/bash
source $(dirname $0)/common

TYPE=dyndata-meta
node_clear $TYPE

./zkctl -w $TYPE add --$TYPE-id=brief    --data='{"target_module":"bc","reload_command":"-t 60"}'
./zkctl -w $TYPE add --$TYPE-id=mcdict   --data='{"target_module":"bc","reload_command":"-c 1 -p 4428"}'
./zkctl -w $TYPE add --$TYPE-id=strategy --data='{"target_module":"bc","reload_command":"-c 2 -p 1234"}'
./zkctl -w $TYPE list | sort


./zkctl -w dyndata add --dyndata-id=error_id --data='{"hello":"hjz"}' 2>&1
./zkctl -w dyndata add --dyndata-id=brief_2012 --data='{"hello":"hjz"}'
./zkctl -w dyndata add --dyndata-id=brief_2013 --data='{"hello":"hjz"}'
#1.data
./zkctl -w dyndata add --dyndata-id=brief_2014 --data='{"hello":"hjz"}'
./zkctl -w dyndata get-meta-all --dyndata-id=brief_2014
#2.default-data
./zkctl -w dyndata add --dyndata-id=brief_2015
./zkctl -w dyndata get-meta-all --dyndata-id=brief_2015
#3.default+kvs
./zkctl -w dyndata add --dyndata-id=brief_2016 --key=a1 --value=b2 --key=a2 --value=b3
./zkctl -s dyndata get-meta-all --dyndata-id=brief_2016
#4.data+kvs
./zkctl -w dyndata add --dyndata-id=brief_2017 --data='{"a":"1"}' --key=b --value=2 --key=c --value=3
./zkctl -s dyndata get-meta-all --dyndata-id=brief_2017

./zkctl -w dyndata-meta list-dyndata --dyndata-meta-id=brief

./zkctl -w dyndata-meta del --dyndata-meta-id=brief 2>&1
./zkctl -w dyndata-meta del --dyndata-meta-id=mcdict

node_clear dyndata
node_clear dyndata-meta
