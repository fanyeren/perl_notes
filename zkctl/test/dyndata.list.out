brief
mcdict
strategy
dyndata "error_id" add failed, no valid dyndata-meta found.
{"hello":"hjz"}
{"reload_command":"-t 60","target_module":"bc"}
{"a1":"b2","a2":"b3","reload_command":"-t 60","target_module":"bc"}
{"a":"1","b":"2","c":"3"}
brief_2017
brief_2016
brief_2015
brief_2014
brief_2013
brief_2012
dyndata-meta del failed, probally has child
